### PR TITLE
DEV: Adopt modern ember primitives

### PR DESCRIPTION
**Pull request by s3lase**
Modernise components and controllers

### CODE REVIEW

File: assets/javascripts/discourse/components/show-user-notes.hbs [hunk 1]
```diff
@@ -1,7 +1,7 @@
-<a {{action show}} href class="btn btn-default">
+<a {{on "click" @show}} href class="btn btn-default">
   {{d-icon "pencil-alt"}}
   {{#if showCount}}
-    {{i18n "user_notes.show" count=count}}
+    {{i18n "user_notes.show" count=@count}}
   {{else}}
     {{i18n "user_notes.title"}}
   {{/if}}
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse/components/show-user-notes.hbs:7]: A little beyond the scope of 'adopt modern ember primatives'... but I wonder, could this be easily converted to a DButton? It feels a little strange that we have an `<a` with no href, and the `btn btn-default` classes.

File: assets/javascripts/discourse/controllers/user-notes.js [hunk 2]
```diff
@@ -1,70 +1,80 @@
...
-  attachDisabled(newNote, saving) {
-    return saving || !newNote || newNote.length === 0;
-  },
+    this.reset();
+  }
 
-  _refreshCount() {
+  #refreshCount() {
     if (this.callback) {
       this.callback(this.get("model.length"));
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse/controllers/user-notes.js:23]:
```suggestion
      this.callback(this.model.length);
```
?

File: assets/javascripts/discourse-user-notes/templates/connectors/admin-user-controls-after/add-user-notes-button.hbs [hunk 3]
```diff
@@ -1 +1 @@
-{{show-user-notes show=(action "showUserNotes") count=userNotesCount}}
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse-user-notes/templates/connectors/admin-user-controls-after/add-user-notes-button.hbs:1]: Looks like there are a couple more connector templates which could be modernised. In particular I'm thinking of the `{{action` modifiers here:

https://github.com/discourse/discourse-user-notes/blob/1fc0b0241f5ecfc94814bd40b02eb316e507e2ac/assets/javascripts/discourse-user-notes/templates/connectors/user-card-post-names/show-user-notes-on-card.hbs#L2

https://github.com/discourse/discourse-user-notes/blob/1fc0b0241f5ecfc94814bd40b02eb316e507e2ac/assets/javascripts/discourse-user-notes/templates/connectors/after-reviewable-post-user/show-user-notes-on-flags.hbs#L2

File: assets/javascripts/discourse/controllers/user-notes.js [hunk 4]
```diff
@@ -1,70 +1,80 @@
...
+
+  constructor() {
+    super(...arguments);
 
-  @discourseComputed("newNote", "saving")
-  attachDisabled(newNote, saving) {
-    return saving || !newNote || newNote.length === 0;
-  },
+    this.reset();
+  }
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse/controllers/user-notes.js:19]: Now that this is a native class, I think we can just initialize the properties when they're initially defined, and get rid of the `reset` method. Looks like all of them get initialized to `null` anyway, so we only need to give `saving` an initial value:

i.e.:

```
@tracked saving = false;
```

File: assets/javascripts/discourse/controllers/user-notes.js [hunk 5]
```diff
@@ -1,70 +1,80 @@
...
-      let args = {
-        raw: this.newNote,
-        user_id: userId,
-      };
+    this.saving = true;
 
-      if (this.postId) {
-        args.post_id = parseInt(this.postId, 10);
-      }
+    let args = {
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse/controllers/user-notes.js:45]: Tiny suggestion:
```suggestion
    const args = {
```

**Review by davidtaylorhq — COMMENTED** [review]
This looks great - thanks @s3lase. Just a few minor comments. We should also add a `.discourse-compatibility` file to make sure this update doesn't get installed on `stable` discourse sites.

Good call on adding some tests as well - this is a pretty popular plugin so they'll be very helpful for preventing future regressions 👌

File: assets/javascripts/discourse/controllers/user-notes.js (hunk 4 above)
Comment (davidtaylorhq) [reply assets/javascripts/discourse/controllers/user-notes.js:19]: Ah I see, I didn't realise it was being called from another place. We'll have to keep it 👌

Comment (s3lase) [reply assets/javascripts/discourse/controllers/user-notes.js:19]: Great!

**Review by davidtaylorhq — APPROVED** [review]
Looks great to me - thanks @s3lase 

If you think the `DButton` suggestion is easily achievable then let's do it. Otherwise, happy to merge as-is.

**Comment by s3lase** [comment]
Thanks @davidtaylorhq 

The `DButton` change is done 

https://github.com/discourse/discourse-user-notes/blob/1dad852b29628def67547a9f516cb4c3a0d9e9ef/assets/javascripts/discourse/components/show-user-notes.hbs#L1-L6

I'm now investigating a possible issue with this connector

https://github.com/discourse/discourse-user-notes/blob/1dad852b29628def67547a9f516cb4c3a0d9e9ef/assets/javascripts/discourse/templates/connectors/admin-dashboard-moderation-bottom/recent-user-notes-report-table.hbs#L1-L7

It appears `lastWeek ` and `endDate` aren't available as used in the template.

**Pull request merged**